maptk/geo_reference_points_io.cxx

```diff
@@ -99,20 +99,20 @@ void load_reference_file(vital::path_t const& reference_file,
     // input landmarks are given in lon/lat/alt format (ignoring alt for now)
     ss >> vec;
 
-    // When this is called the first time, setzone is given a -1, which is the
-    // default for the function.
-    lgcs.geo_map_algo()->latlon_to_utm(vec.y(), vec.x(), x, y, zone, northp,
-                                       lgcs.utm_origin_zone());
-    vec[0] = x; vec[1] = y; vec[2] = vec.z();
-    mean += vec;
-
-    // Use the zone of the first input landmark as the base zone from which we
-    // interpret all other geo-positions with respect to.
-    if (set_lgcs_origin)
+    vital::vector_2d lon_lat(vec.x(), vec.y());
+    vital::geo_point gp(lon_lat, vital::SRID::lat_lon_WGS84);
+    if ( lgcs.origin().is_empty() )
     {
-      LOG_DEBUG(logger, "lgcs origin zone: " << zone );
-      lgcs.set_utm_origin_zone(zone);
+      auto zone = vital::utm_ups_zone( lon_lat );
+      crs = (zone.north ? vital::SRID::UTM_WGS84_north : vital::SRID::UTM_WGS84_south) + zone.number;
+      LOG_DEBUG(logger, "lgcs origin zone: " << zone.number );
+      lgcs.set_origin( vital::geo_point( gp.location( crs ), crs ) );
```

Comment:
This seems to be doing extra work, since `set_origin` will also convert to UTM. Maybe just call `set_origin` with the lat/lon point, then do `crs = lgcs.origin().crs();`?